packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -434,10 +434,18 @@ mixin SchedulerBinding on BindingBase {
   }
 
   // Scheduled by _ensureEventLoopCallback.
-  void _runTasks() {
+  Future<void> _runTasks() async {
     _hasRequestedAnEventLoopCallback = false;
     if (handleEventLoopCallback())
       _ensureEventLoopCallback(); // runs next task when there's time
+    else {
+      if (_taskQueue.isNotEmpty) {
+        // Avoid continuously adding tasks to the underlying queue
+        await endOfFrame;
```

Comment:
This has the side effect of potentially scheduling a frame if none is underway right now. That may be undesirable?

Reply:
Yeah, it doesn't seem to work that way